### PR TITLE
Pjtester support inrepo

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

1. Added support for inrepo config. pjtester can load presubmits and postsubmits defined as inrepo config.
2. pjtester can be triggered on pr on any repo. pjtester can run test prowjob defined for any repository.
3. Added feature of providing pull request number with prowjob definition to use when creating test prowjob.
4. pjtester add its prefix to github context created when running test prowjob against some pull request. This prevents overriding existing context status.
5. Removed support for loading prow config from custom path provided in pjtester.yaml file. Prow config for test prowjob must be placed in our standard location.
6. Removed support for loading jobs config from custom path provided in pjtester.yaml file. Jobs config for test prowjob must be placed in our standard location or in standard location for inrepo config.
7. Added debug messages.
8. Refactored code.
9. Added validation of values provided in pjtester.yaml file.
10. Replaced test. Using ginkgo and gomega libraries for tests. Using more behavioural like tests.

Fixes: #6156

🔥 **BREAKING CHANGE**🔥

### CODE REVIEW

**Review by Ressetkk — COMMENTED** [review]
One comment, otherwise LGTM

**Comment by dekiel** [comment]
pre-main-test-infra-lint job failed again when loading files to scan for path ./development/gcp/cloud-functions/serviceaccountcleaner
/override pre-main-test-infra-lint

**Review by Ressetkk — APPROVED** [review]
I'll exceptionally approve it. Let's not block it. However if by any chance you hit this function, your entire test execution will crash. Ask yourself if you want to write code that way and keep unsafe crashes in your functions.

**Comment by dekiel** [comment]
If a test will reach that function and all tests will crash that's perfectly fine. I will know then that I need to implement that function. At present this function is not used. This crash is not unsafe. It will not crash production system. Data will not leak. Server will not hang. Memory will not leak. Threads will not leak. It will stop test with information why and where and this is exactly what that function should do now.

**Comment by kasiakepka** [comment]
@dekiel can you please look sat that failing job?

File: development/github/pkg/client/v2/client.go [hunk 1]
```diff
@@ -35,9 +40,12 @@ func (o *GithubClientConfig) NewGithubClient(options ...GithubClientOption) (*Gi
...
 		return nil, err
 	}
-	return &GithubClient{
-		Client: client,
-	}, nil
+	/*
+		return &GithubClient{
+			Client: client,
+		}, nil
+	*/
```
Comment (Halamix2) [inline development/github/pkg/client/v2/client.go:47]: Is this commented out code needed?

File: development/tools/pkg/pjtester/pjtester.go [hunk 2]
```diff
@@ -100,333 +133,298 @@ func checkEnvVars(varsList []string) error {
...
+	if err != nil {
+		// this check is only needed when your code could produce
+		// an invalid value for validation such as interface with nil
+		// value most including myself do not usually have code like this.
+		if _, ok := err.(*validator.InvalidValidationError); ok {
+			fmt.Println(err)
+			return testCfg{}, err
 		}
+		// from here you can create your own error messages in whatever language you wish
+		return testCfg{}, err
```
Comment (Halamix2) [inline development/tools/pkg/pjtester/pjtester.go:175]: Do we need to differentiate between error types just to Print error in this function? Shouldn't we just return err no matter the type if we don't do any operations on it based on type?

File: development/tools/pkg/pjtester/pjtester.go [hunk 3]
```diff
@@ -442,57 +440,226 @@ func formatPjName(pullAuthor, pjName string) string {
...
 		pj.Spec.Report = true
 	} else {
 		pj.Spec.ReporterConfig = &prowapi.ReporterConfig{Slack: &prowapi.SlackReporterConfig{Channel: "kyma-prow-dev-null"}}
 	}
-	return pj
+	return pj, nil
 }
 
-// SchedulePJ will generate prowjob for testing and schedule it on prow for execution.
+// SchedulePJ will generate ProwJob for testing and crate it on prow k8s cluster for execution.
```
Comment (Halamix2) [inline development/tools/pkg/pjtester/pjtester.go:570]:
```suggestion
// SchedulePJ will generate ProwJob for testing and create it on prow k8s cluster for execution.
```

File: development/github/pkg/client/v2/client.go (hunk 1 above)
Comment (dekiel) [reply development/github/pkg/client/v2/client.go]: Removed.

File: development/tools/pkg/pjtester/pjtester.go (hunk 2 above)
Comment (dekiel) [reply development/tools/pkg/pjtester/pjtester.go]: Not needed, removed.

File: development/tools/pkg/pjtester/pjtester.go (hunk 3 above)
Comment (dekiel) [reply development/tools/pkg/pjtester/pjtester.go]: Corrected.

**Pull request merged**